galasa-managers-parent/galasa-managers-comms-parent/dev.galasa.mq.manager/src/main/java/dev/galasa/mq/internal/MQManagerImpl.java

```diff
@@ -0,0 +1,253 @@
+		} catch (ConfigurationPropertyStoreException e1) {
+			throw new MqManagerException("Unable to access framework services", e1); 	
+		}
+        if(galasaTest.isJava()) {
+            List<AnnotatedField> ourFields = findAnnotatedFields(MqManagerField.class);
+            if (!ourFields.isEmpty()) {
+                youAreRequired(allManagers, activeManagers, galasaTest);
+            }
+        }
+        
+        this.storedArtifactsRoot = getFramework().getResultArchiveStore().getStoredArtifactsRoot();
+        try {
+			this.credentialService = getFramework().getCredentialsService();
+		} catch (CredentialsException e) {
+			throw new MqManagerException("Unable to access credentials service",e);
+		}
+    }
+
+    @Override
+    public void youAreRequired(@NotNull List<IManager> allManagers, @NotNull List<IManager> activeManagers, @NotNull GalasaTest galasaTest)
+            throws ManagerException {
+        if (activeManagers.contains(this)) {
+            return;
+        }
+        activeManagers.add(this);
+    }
+    
+    @Override
+    public boolean doYouSupportSharedEnvironments() {
+        return true;   // this manager does not provision resources, therefore support environments 
+    }
+    
+    @Override
+    public void provisionStart() throws ManagerException, ResourceUnavailableException {
+    	/*
+    	 * This method of starting the qmgrs and then the queues is not the most effective
+    	 * If we had a test with lots of qmgrs and queues then it would be better to 
+    	 * start the qmgrs async and on completion start the required queues, however this 
+    	 * is unlikely to be a real issue
+    	 */
+    	//start the queue managers first 
+    	for(Entry<String, MessageQueueManagerImpl> entry : this.queueManagers.entrySet()) {
+    		MessageQueueManagerImpl qmgr = entry.getValue();
+    		try {
+    			qmgr.startup();
+    		}catch(MqManagerException mqme) {
+    			logger.error("Unable to start Queue Manager");
+    			throw mqme;
+    		}	
+    	}
+    	
+    	//Now that the queue managers are active start the queues
+    	for(MessageQueueImpl queue : this.queues) {
+    		queue.startup();
+    	}
+    	
+    }
+    
+    @Override
+    public void provisionStop() {
+    	for(Entry<String, MessageQueueManagerImpl> entry : this.queueManagers.entrySet()) {
+    		MessageQueueManagerImpl qmgr = entry.getValue();
+    		qmgr.close();	
+    	}
+    }
+    
+    @Override
+    public void startOfTestMethod(@NotNull GalasaMethod galasaMethod) throws ManagerException {
+    	super.startOfTestMethod(galasaMethod);
+    	this.currentMethod = galasaMethod.getJavaExecutionMethod().getName();
+    }
+
+	public String getCurrentMethod() {
+		return currentMethod;
+	}
+	
+	public ICredentials getCredentials(String tag) throws MqManagerException {
+		//pull the access credentials
+    	ICredentials credentials;
+    	String credentialsKey = NAMESPACE+"."+getInstanceForTag(tag);
```

Comment:
The CPS properties for the for the qmgr needs to provide the Credentials ID,   instead of forcing it to be the instance id.   This is so two qmgrs,  or other resources can share the same credentials,  so at password change day,  you are not unnecessarily changing multiple credentials.

Reply:
The default credentals ID can default to the namespace.

Reply:
see dev.galasa.openstack.manager.internal.properties.OpenStackCredentialsId